.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,73 @@
+name: Unit tests
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./tests
+    
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        submodules: 'recursive'
+    
+    - name: Unpack test files
+      run: |
+        tar xf elasticc2_alert_test_data.tar.bz2
+    
+    - name: Build docker images
+      run: |
+        docker compose build
+    
+    - name: Bring docker environments up
+      run: |
+        docker compose up -d shell
+    
+    - name: Run Cycle Tests
+      run: |
+        docker compose exec -it shell /bin/bash -c "python3 -m pytest /tests/test_elasticc2_spectrumcycle.py"
```

Comment:
I would also add test_elasticc2_alertcycle.py and test_elasticc2_spectrumcycle.py to the tests that are done here.  I'm not sure that all the other tests are up to date (I need to get on that!), but those two should be.

Really, what I should do is clean up the test directory and put tests in subdirectories, so that the ones we want run in GH actions are collected together in some subdirectories, and one-offs that we run manually are elsewhere.